src/vfs.c

```diff
@@ -340,3 +340,27 @@ esp_err_t frogfs_vfs_register(const frogfs_vfs_conf_t *conf)
     s_frogfs_vfs[index] = vfs;
     return ESP_OK;
 }
+
+esp_err_t frogfs_vfs_deregister(const char *base_path)
+{
+    assert(base_path != NULL);
+
+    for (int i = 0; i < CONFIG_FROGFS_MAX_PARTITIONS; i++)
+    {
+        if (s_frogfs_vfs[i] != NULL)
+        {
```

Comment:
A trick I've learned over the years to reduce the nesting of your statements is to reverse the condition, and use continue.  It is just as efficient but provides easier to read code.

```c
if (s_frogfs_vfs[i] == NULL) {
    continue;
}
```